charts/argo-cd/templates/application-controller/deployment.yaml

```diff
@@ -0,0 +1,287 @@
+{{- if .Values.controller.deployment }}
+apiVersion: apps/v1
+kind: Deployment
```

Comment:
Instead of having a whole file, would it not be easier to change the `kind` based on the `{{- if .Values.controller.deployment }}` ?

Reply:
I agree it would be much easier.  However I wanted to have a clear separation between the `Deployment` & `StatefulSet`, specifically if the number of `replicas` was increased beyond 1 and `dynamicClusterDistribution` was not enabled.

I'm happy to make the suggested changes if that makes more sense.